pkg/build/controller/common/util.go

```diff
@@ -134,3 +140,48 @@ func HasBuildPodNameAnnotation(build *buildapi.Build) bool {
 	_, hasAnnotation := build.Annotations[buildapi.BuildPodNameAnnotation]
 	return hasAnnotation
 }
+
+// ProcessValueFrom supports the k8s pod syntax of allowing "valueFrom" in addition
+// to "value", where "valueFrom" uses the downward api to reference existing env
+// variables, jsonpath references to the build object, secrets, and configmaps.
+func ProcessValueFrom(pod *kapi.Pod, client kclientset.Interface) error {
+	var outputEnv []kapi.EnvVar
+	var allErrs []error
+
+	build, version, err := buildadmission.GetBuildFromPod(pod)
```

Comment:
I guess the answer to that may be, excessive work for this PR but worth scheduling for tech debt?